qmp/socket.go

```diff
@@ -68,6 +68,26 @@ func NewSocketMonitor(network, addr string, timeout time.Duration) (*SocketMonit
 	return mon, nil
 }
 
+// NewListeningSocketMonitor listens for a monitor socket connection from Qemu.
+// An error is returned if the connection fails to accept.
+func NewListeningSocketMonitor(network, addr string) (*SocketMonitor, error) {
+	listener, listenerErr := net.Listen(network, addr)
+	if listenerErr != nil {
+		return nil, listenerErr
+	}
+	c, err := listener.Accept()
```

Comment:
defer listener.Close()

Reply:
I've added defer. Connection is working as expected.